community/eigen/APKBUILD

```diff
@@ -1,36 +1,51 @@
+# Contributor: stef <l0ls0fo2i@ctrlc.hu>
 # Contributor: Bradley J Chambers <brad.chambers@gmail.com>
 # Maintainer: Bradley J Chambers <brad.chambers@gmail.com>
 pkgname=eigen
 pkgver=3.3.7
-pkgrel=0
+pkgrel=1
 pkgdesc="Eigen is a C++ template library for linear algebra"
 url="http://eigen.tuxfamily.org/index.php?title=Main_Page"
 arch="noarch"
-options="!check"  # Headers-only: no tests are possible.
 license="MPL-2.0"
 depends=""
-makedepends=""
+options="!check"  # checks take a long time, and tend to fail randomly
+makedepends="cmake openblas-dev suitesparse-dev mpfr-dev fftw-dev boost-dev gmp-dev glu-dev freeglut-dev"
 install=""
 subpackages="$pkgname-dev $pkgname-doc"
-source="$pkgname-$pkgver.tar.gz::http://bitbucket.org/eigen/$pkgname/get/$pkgver.tar.gz"
+source="$pkgname-$pkgver.tar.gz::http://bitbucket.org/eigen/$pkgname/get/$pkgver.tar.gz
+	fftw-test-fix.patch"
 builddir="$srcdir/$pkgname-$pkgname"
 
 build() {
-        return 0
+	mkdir -p build
+	cd build
+	# needs -std so c11 tests are being built
+	CXXFLAGS="-std=c++11" \
+	cmake -DCMAKE_INSTALL_PREFIX=/usr "$builddir"
+	make
 }
 
 prepare() {
-        mv "$builddir"-* "$builddir"  # directory name contains hash
-        default_prepare
+	mv "$builddir"-* "$builddir"  # directory name contains hash
+	default_prepare
+}
+
+check() {
+	cd "$builddir/build"
+	make check
 }
 
 package() {
-        mkdir -p "$pkgdir"/usr/include/eigen3
-        cp -r "$builddir"/Eigen "$pkgdir"/usr/include/eigen3
-        cp -r "$builddir"/unsupported "$pkgdir"/usr/include/eigen3
-        cp "$builddir"/signature_of_eigen3_matrix_library "$pkgdir"/usr/include/eigen3
-        mkdir -p "$pkgdir"/usr/share/doc/$pkgname
-        cp -r "$builddir"/doc/ "$pkgdir"/usr/share/doc/$pkgname
+	cd "$builddir/build"
+	make DESTDIR="$pkgdir" install
+	mkdir -p "$pkgdir"/usr/share/doc/$pkgname
+	cp -r "$builddir"/doc/ "$pkgdir"/usr/share/doc/$pkgname
+	mkdir -p ${pkgdir}-dev/usr/share/cmake/Modules
+	mv $pkgdir/usr/share/eigen3/cmake/*.cmake ${pkgdir}-dev/usr/share/cmake/Modules
```

Comment:
Doesn't the -dev package do this automatically? If not, this is the wrong approach: You'd do a `dev() {}` function, which runs `default_dev` and then installs these files into `"$subpkgdir"`, not `$pkgdir-dev`.

Reply:
aaah. thanks! will do that.

Reply:
done. seems to work as expected (locally).